### DISCUSSION TITLE
Getting All Instructions Within a Function

### DISCUSSION THREAD

**Discussion by rstone-ws** [NationalSecurityAgency/ghidra#6078] [Q&A] [answered]
I'm writing a plugin and need to find a way to get all termination instructions in a function. Is there a way to get instructions from only a specific function? I've found a number of ways to iterate through instructions forward from an address, but need a sane way to know where the function ends (i.e. do not iterate to addresses outside the function).
Is there a built-in that can get all instructions from a function? Or any suggestions on how to do this? 
Thanks!

**Comment by ryanmkurtz** [comment, marked as answer]
Can you do `function.getBody()` and go from there?

**Reply by rstone-ws** [reply]
Ah! That should be enough to get it going, I knew I had to be missing something. Thanks a bunch.

**Comment by dev747368** [comment]
Have you looked at using BlockModel?  Take a look at ```CyclomaticComplexity``` as an example of using ```BasicBlockModel``` to break a function up into blocks of instructions and iterating over them, detecting references, if the block terminates, etc.
